File: cpp_tests/test_datum.cpp
#include "tests/test_datum.hpp"

void TestDatum() {
  Datum datum1{1, 1, 2025};
  assert(datum1.GetTag() == 1);
  assert(datum1.GetMonat() == 1);
  assert(datum1.GetJahr() == 2025);
  assert(datum1.Print() == "01.01.2025");

  Datum datum2{2, 2, 2024};
  assert(datum2.GetTag() == 2);
  assert(datum2.GetMonat() == 2);
  assert(datum2.GetJahr() == 2024);
  assert(datum2.Print() == "02.02.2024");

  assert(datum1.operator==(datum2) == false);

  std::cout << "Alle Tests für Datum waren erfolgreich!" << std::endl;
}

File: python_tests/test_datum.py
"""Python-file for testing class Datum"""

from py_bindings import Datum

def test_datum():
    """Funktion zum Testen von Datum"""
    datum1 = Datum(1, 1, 2025)
    datum2 = Datum(10, 12, 2022)

    assert datum1.get_tag() == 1
    assert datum1.get_monat() == 1
    assert datum1.get_jahr() == 2025
    assert datum1.datum_anzeige() == "01.01.2025"
    assert datum1.vergleiche_datum(datum2) is False
    assert str(datum2) == f"Datum: {datum2.datum_anzeige()}"

test_datum()